registry/data-models/common/models.py

```diff
@@ -0,0 +1,139 @@
+
+
+class TensorFeatureFormat(BaseModel):
+    """
+    Defines the format of feature data. Feature data is produced by applying transformation on source, in a Feature.
+    Tensor is used to represent feature data. A tensor is a generalization of vectors and matrices to potentially
+    higher dimensions.
+    """
+    tensorCategory: TensorCategory  # Type of the tensor.
+    valueType: ValueType  # Type of the value column.
+    dimensions: List[Dimension]  # A feature data can have zero or more dimensions (columns that represent keys).
+
+
+class FeatureType(BaseModel):
+    """
+   Information about a featureName. It defines the type, format and default value.
+   Tensor is the next generation representation of the features, so using
+   Tensor type w TensorFeatureFormat would be preferable FeatureType.
+    """
+    type: FeatureValueType  # Defines the high level semantic type of feature.
+    format: Optional[TensorFeatureFormat]  # Defines the format of feature data.
+    defaultValue: Union[bool, int, float, str, bytes]
+
+
+class Clazz(BaseModel):
+    """
+    Reference to a class by fully-qualified name
+    """
+    fullyQualifiedName: str  # A fully-qualified class name including paths.
+
+
+class Function(BaseModel):
+    """
+    Base model for all functions
+    """
+    expression: str  # Expression in str format
+
+
+class MvelExpression(Function):
+    """
+    An expression in MVEL language.
+    """
+    mvel: str  # The MVEL expression
+
+
+class UserDefinedFunction(Function):
+    """
+    User defined function that can be used in feature extraction or derivation.
+    """
+    clazz: Clazz  # Reference to the class that implements the user defined function.
+    parameters: Dict[str, json] = {}  # This field defines the custom parameters of the user defined function
+
+
+class SparkSqlExpression(Function):
+    """
+    An expression in Spark SQL.
+    """
+    sql: str  # Spark SQl expression
+
+
+class SemanticVersion(BaseModel):
+    """
+    A representation of a semantic version (see https://semver.org/)
+    """
+    majorVersion: int  # The major version of this version. This is the x in x.y.z.
+    minorVersion: int  # The minor version of this version. This is the y in x.y.z
+    patchVersion: int  # The patch version of this version. This is the z in x.y.z
+    metadata: Optional[str]  # Optional build metadata attached to this version.
+
+
+class FeathrModel(BaseModel):
+    """
+    Base model for feathr entity which will be displayed in Feathr UI
+    """
+    displayName: str  # name of the entity showed on UI
+    typeName: str  # type of entity in str format, will be displayed in UI
+    # Represents zero, one or multiple keyPlaceholderRefs which are used as the
+    # identifiers to reference KeyPlaceholders of the FeatureSource
+    keyPlaceholderRefs: List[str] = []
```

Comment:
Do we need this as a common field? Feels like this should be a more specific field to some entity